pandas/core/groupby/grouper.py

```diff
@@ -856,3 +821,25 @@ def _convert_grouper(axis: Index, grouper):
         return grouper
     else:
         return grouper
+
+
+def _check_deprecated_resample_kwargs(kwargs):
+    if kwargs.get("base", None) is not None:
+        warnings.warn(
+            "'base' in .resample() and in Grouper() is deprecated.\n"
+            "The new arguments that you should use are 'offset' or 'origin'.\n"
+            '\n>>> df.resample(freq="3s", base=2)\n'
+            "\nbecomes:\n"
+            '\n>>> df.resample(freq="3s", offset="2s")\n',
+            FutureWarning,
```

Comment:
Passing no stacklevel is basically never the correct thing to do. I understand it is difficult to set the correct stacklevel for all code paths that get here, but then I would at least choose one code path, and determine the stacklevel for that one.

Reply:
There is a(t least one) solution to reliably ensure the correct stacklevel everywhere (#36629) but it seems too complicated to be worthwhile (see the discussion there). 

On the other hand, ensuring the correct stacklevel for only one path would produce correct warnings for some calls, but may produce completely misleading warnings for other paths (e.g. it could "overshoot" the stacklevel into the user's code). IMO this is worse than having no stacklevel handling at all.

Reply:
When specifying no stacklevel, then it actually uses the default of `stacklevel=1`, which *also* gives misleading warnings from where it coming from, I think? So IMO better to have part of the warning not be misleading, than have all of them misleading.

Reply:
I wouldn't say that the default is misleading, since it just uses the code that actually raised the warning. In this case it would be `_check_deprecated_resample_kwargs`. Sure, it might not be the most informative, but I wouldn't say that's misleading. On the other hand, setting to high a stacklevel (which could happen if we only take care of a couple of paths) could very possibly set the stacklevel to user code like `main`, the top-level module code, or even in an extreme case the interpreter (so it would show `sys`) .

CC @jreback, any opinion?

Reply:
Reverted in e842308.